extensions/amp-story/1.0/amp-story-page.js

```diff
@@ -563,8 +566,11 @@ export class AmpStoryPage extends AMP.BaseElement {
     const audioEl = upgradeBackgroundAudio(this.element, loop);
     if (audioEl) {
       this.mediaPoolPromise_.then((mediaPool) =>
-        this.registerMedia_(mediaPool, dev().assertElement(audioEl)).then(() =>
-          mediaPool.preload(dev().assertElement(audioEl))
+        this.registerMedia_(mediaPool, dev().assertElement(audioEl)).then(
+          () => {
+            mediaPool.preload(dev().assertElement(audioEl));
+            this.backgroundAudioDeferred_.resolve();
```

Comment:
Woah actually thanks to the new `waitForPlaybackMediaLayout_()` method we can do something cool.

`waitForPlaybackMediaLayout_` will wait for all media to be laid out, and then it'll register/preload them. With this PR you're making `waitForPlaybackMediaLayout_` wait for the background-audio as well, so there's no need to register/preload them manually like we're doing here. We can delete this block 🥳 

```
const audioEl = upgradeBackgroundAudio(this.element, loop);
this.backgroundAudioDeferred_.resolve();
```

and that's it :))

I like how there's now only one flow to layout/register/preload media elements. No more visiblityupdates etc